core/server/services/auth/api-key/admin.js

```diff
@@ -0,0 +1,112 @@
+const jwt = require('jsonwebtoken');
+const models = require('../../../models');
+const common = require('../../../lib/common');
+
+const JWT_OPTIONS = {
+    maxAge: '5m',
+    algorithms: ['HS256']
+};
+
+/**
+ * Remove 'Ghost' from raw authorization header and extract the JWT token.
+ * Eg. Authorization: Ghost ${JWT}
+ * @param {string} header
+ */
+const _extractTokenFromHeader = function extractTokenFromHeader(header) {
+    const [scheme, token] = header.split(' ');
+
+    if (/^Ghost$/i.test(scheme)) {
+        return token;
+    }
+
+    return;
+};
+
+/**
+ * Admin API key authentication flow:
+ * 1. extract the JWT token from the `Authorization: Ghost xxxx` header
+ * 2. decode the JWT to extract the api_key id from the "key id" header claim
+ * 3. find a matching api_key record
+ * 4. verify the JWT (matching secret, matching URL path, not expired)
+ * 5. place the api_key object on `req.api_key`
+ *
+ * There are some specifcs of the JWT that we expect:
+ * - the "Key ID" header parameter should be set to the id of the api_key used to sign the token
+ *   https://tools.ietf.org/html/rfc7515#section-4.1.4
+ * - the "Audience" claim should match the requested API path
+ *   https://tools.ietf.org/html/rfc7519#section-4.1.3
+ */
+const authenticateAdminApiKey = function authenticateAdminApiKey(req, res, next) {
```

Comment:
Do we still need this logic? If we only support content api keys for now?

Reply:
I think it's best we leave this in, as we haven't come to 100% decision yet. We can remove the adminapikey stuff from the auth middleware if we decide to go without it :+1:

Reply:
more like my bad! we have :) I'll split the admin auth into a separate PR :relaxed: